src/audio_worklet.js

```diff
@@ -0,0 +1,168 @@
+// This file is the main bootstrap script for Wasm Audio Worklets loaded in an Emscripten application.
+// Build with -sAUDIO_WORKLET=1 linker flag to enable targeting Audio Worklets.
+
+// AudioWorkletGlobalScope does not have a onmessage/postMessage() functionality at the global scope, which
+// means that after creating an AudioWorkletGlobalScope and loading this script into it, we cannot
+// postMessage() information into it like one would do with Web Workers.
+
+// Instead, we must create an AudioWorkletProcessor class, then instantiate a Web Audio graph node from it
+// on the main thread. Using its message port and the node constructor's
+// "processorOptions" field, we can share the necessary bootstrap information from the main thread to
+// the AudioWorkletGlobalScope.
+
+function createWasmAudioWorkletProcessor(audioParams) {
+  class WasmAudioWorkletProcessor extends AudioWorkletProcessor {
+    constructor(args) {
+      super();
+
+      // Copy needed stack allocation functions from the Module object
+      // to global scope, these will be accessed in hot paths, so maybe
+      // they'll be a bit faster to access directly, rather than referencing
+      // them as properties of the Module object.
+      globalThis.stackAlloc = Module['stackAlloc'];
+      globalThis.stackSave = Module['stackSave'];
+      globalThis.stackRestore = Module['stackRestore'];
+      globalThis.HEAPU32 = Module['HEAPU32'];
+      globalThis.HEAPF32 = Module['HEAPF32'];
+
+      // Capture the Wasm function callback to invoke.
+      let opts = args.processorOptions;
+      this.callbackFunction = Module['wasmTable'].get(opts['cb']);
```

Comment:
Thank you very much for the PR!

I am testing it as an output module for vlc, and the issue I have is that I don't understand why Module.wasmTable, is undefined.

I am wrapping the callback chain in an [Init](https://code.videolan.org/b1ue/vlc/-/blob/em_aout/1/modules/audio_output/emscripten.c#L301) function that I proxy to the main thread.

I assumed globalThis.Module was not defined because the loading is asynchronous, but it seems okay when I log it (cf. [screenshot](https://imgur.com/a/7bW5RZJ)).

Another issue I have is I want to block the calling thread to wait for the Init() to actually finish the async chain (that ends with .connect(), or .resume() eventually), maybe the problem is that I am not doing it correctly [here](https://code.videolan.org/b1ue/vlc/-/blob/em_aout/1/modules/audio_output/emscripten.c#L247)?

Reply:
after another pass, it seems this :
```
onmessage = function(d) {
        // The first message sent to the Worker is always the bootstrap message.    
        // Drop this message listener, it served its purpose of bootstrapping       
        // the Wasm Module load, and is no longer needed. Let user code register    
        // any desired message handlers from now on.                                
        onmessage = null;
        d = d.data;
        self.Module = d;
        d['instantiateWasm'] = (info, receiveInstance) => { var instance = new WebA\
ssembly.Instance(d['wasm'], info); receiveInstance(instance, d['wasm']); return ins\
tance.exports; }
        importScripts(d.js);
        // Drop now unneeded references to from the Module object in this Worker,   
        // these are not needed anymore.                                            
        d.wasm = d.mem = d.js = 0;
    console.error("DEBUG:", Module["wasmTable"]);
}
```
is not called at all, which could explain why, the Module object in the worklet is not filled correctly. wasmTable is not the only Module property I am missing.

Reply:
`_wasm_worker_initializeRuntime` is not called, but then how is it that the tone-generator test runs without issues? Sorry, I am missing something.

Reply:
Confirmed, totally my fault, I must've missed this message :
```
Aborted(Assertion failed: Pthreads do not work in this environment yet (need Web Workers, or an alternative to them))
```